sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SortOrder.scala

```diff
@@ -28,14 +28,24 @@ abstract sealed class SortDirection {
   def sql: String
 }
 
+// default null order is firt for asc
 case object Ascending extends SortDirection {
   override def sql: String = "ASC"
 }
 
+case object AscendingNullLast extends SortDirection {
```

Comment:
Thanks! I think this could work.
```
case class SortOrder(child: Expression, direction: SortDirection, nullOrdering: NullOrdering) {
  ...
}

object SortOrder {
  def apply(child: Expression, direction: SortDirection): SortOrder = {
    new SortOrder(child, direction, direction.defaultNullOrdering)
  }
}
```
Let me run the test and see.